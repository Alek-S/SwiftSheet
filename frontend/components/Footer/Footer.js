import React from 'react';
import styled from 'styled-components';
import defaultStyle from '../../defaultStyle';
import { Link } from 'react-router-dom';

/** @function
 * @name Footer
 * Bottom footer element of page - always viewable
 *
 * @returns {JSX}
 */
export default function Footer() {
	return (
		<Styledfooter>
			<div>
				Created with üç∑ by{' '}
				<a href="https://www.alekshnayder.com/" target="_blank" rel="noopener">
					Alek Shnayder
				</a>
			</div>
			<Spacer>|</Spacer>
			<div>
				<IconLink
					style={{ marginRight: '1rem' }}
					href="mailto:swiftsheetapp@example.com?Subject=Hello"
					target="_top"
					rel="noopener"
				>
					<img src="./assets/images/mail.svg" alt="logo" height="22" />
				</IconLink>
				<IconLink
					href="https://twitter.com/SwiftSheetApp"
					target="_blank"
					rel="noopener"
				>
					<img src="./assets/images/twitter.svg" alt="logo" height="25" />
				</IconLink>
			</div>
			<Spacer>|</Spacer>
			<div>
				<Link to="/privacy">Privacy Policy</Link>
			</div>
			<Spacer>|</Spacer>
			<div>
				<Link to="/terms">Terms of Service</Link>
			</div>
		</Styledfooter>
	);
}

const Styledfooter = styled(defaultStyle)`
	box-sizing: border-box;
	padding: 1rem;
	margin-top: 0.5rem;
	text-align: center;
	width: 100%;
	display: flex;
	flex-direction: row;
	flex-wrap: wrap;
	justify-content: center;

	& a {
		color: ${props => props.theme.color.text};
	}

	@media (max-width: 500px) {
		flex-direction: column;
		div {
			margin-top: 0.5rem;
			margin-bottom: 0.5rem;
		}
	}
`;

const Spacer = styled.span`
	margin-left: 1rem;
	margin-right: 1rem;

	@media (max-width: 500px) {
		display: none;
	}
`;

const IconLink = styled.a`
	position: relative;
	top: -5px;

	@media (max-width: 500px) {
		top: 0;
	}
`;

export { Styledfooter };
