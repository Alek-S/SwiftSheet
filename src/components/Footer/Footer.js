import React from 'react';
import styled from 'styled-components';


/** @function 
 * @name Footer
 * Bottom footer element of page - always viewable
 * 
 * @returns {JSX}
*/
export default function Footer() {
	return(
		<Styledfooter>
			Created by <a href='https://www.alekshnayder.com/' target="_blank" rel="noopener">Alek Shnayder</a>
		</Styledfooter>
	);
}

const Styledfooter = styled.footer`
	color: ${props => props.theme.color.text};
	width: 100%;
	background-color: white;
	padding: 1rem;
	text-align: center;
	font-size: .9em;
	font-family: ${props => props.theme.font.main};

	& a{
		color: ${props => props.theme.color.text};
	}
`