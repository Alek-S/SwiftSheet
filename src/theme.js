/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
