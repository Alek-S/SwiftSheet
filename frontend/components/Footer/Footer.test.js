import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';

import Footer, { Styledfooter } from './Footer';
import theme from '../../theme';

describe('Footer', () => {
	test('matches the snapshot', () => {
		const tree = shallow(<Footer theme={theme} />);
		expect(toJson(tree)).toMatchSnapshot();
	});
});

describe('Styledfooter', () => {
	const tree = renderer.create(<Styledfooter theme={theme} />).toJSON();

	test('matches the snapshot', () => {
		expect(tree).toMatchSnapshot();
	});
});
