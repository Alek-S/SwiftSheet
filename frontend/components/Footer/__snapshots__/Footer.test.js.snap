// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches the snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  padding: 1rem;
  margin-top: 0.5rem;
  text-align: center;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 a {
  color: #404040;
}

@media (max-width:500px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c0 div {
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
  }
}

<defaultStyle
  className="c0"
>
  <div>
    Created with üç∑ by
     
    <a
      href="https://www.alekshnayder.com/"
      rel="noopener"
      target="_blank"
    >
      Alek Shnayder
    </a>
  </div>
  <Footer__Spacer>
    |
  </Footer__Spacer>
  <div>
    <Footer__IconLink
      href="mailto:swiftsheetapp@example.com?Subject=Hello"
      rel="noopener"
      style={
        Object {
          "marginRight": "1rem",
        }
      }
      target="_top"
    >
      <img
        alt="logo"
        height="22"
        src="./assets/images/mail.svg"
      />
    </Footer__IconLink>
    <Footer__IconLink
      href="https://twitter.com/SwiftSheetApp"
      rel="noopener"
      target="_blank"
    >
      <img
        alt="logo"
        height="25"
        src="./assets/images/twitter.svg"
      />
    </Footer__IconLink>
  </div>
  <Footer__Spacer>
    |
  </Footer__Spacer>
  <div>
    <Link
      replace={false}
      to="/privacy"
    >
      Privacy Policy
    </Link>
  </div>
  <Footer__Spacer>
    |
  </Footer__Spacer>
  <div>
    <Link
      replace={false}
      to="/terms"
    >
      Terms of Service
    </Link>
  </div>
</defaultStyle>
`;
